Cannot run program "pg_dump" when upgrading cloudbeaver-ce from 25.2.1 to 25.2.2

### Description

Hi everyone,

I have tried to upgrade our Cloudbeaver-ce that we are running on Red Hat Openshift for testing purposes from v25.2.1 to v25.2.2 but the main container fails to load because it can't manage to perform the backup of its database. The previous version had been running fine with no issues.

`
...
16-10-2025 09:36:39.151 [main] DEBUG i.c.service.security.db.CBDatabase - 	Initiate connection pool with management database (PostgreSQL; jdbc:postgresql://postgres:5432/cloudbeaver)
16-10-2025 09:36:39.346 [main] INFO  o.j.d.e.p.b.JDBCDatabasePostgresBackupHandler - Command started: pg_dump cloudbeaver --host postgres --port 5432 --blobs --verbose --file /opt/cloudbeaver/workspace/backup/cloudbeaver_backup_version_cb25.zip --schema cb
16-10-2025 09:36:39.348 [main] ERROR o.j.d.e.p.b.JDBCDatabasePostgresBackupHandler - Create backup is failed: Cannot run program "pg_dump": error=2, No such file or directory
16-10-2025 09:36:39.348 [main] WARN  o.j.d.m.sql.schema.SQLSchemaManager - CB_CE migration has been rolled back
16-10-2025 09:36:39.349 [main] ERROR io.cloudbeaver.server.CBApplication - Error initializing database
org.jkiss.dbeaver.DBException: Error updating management database schema
....
`

I imagine that the pg_dump is already present in the cloudbeaver:25.2.2 image.

### Steps to reproduce

1. I have set the _imageTag_ variable value as "25.2.2"  in the _values.yaml_ file of Cloudbeaver's Helm chart
2. Pushed the new chart to OpenShift
3. The previous container gets deleted and the updated one gets created but fails to start

Here is the container's log
[cloudbeaver.log](https://github.com/user-attachments/files/22946129/cloudbeaver.log)

### Expected/Desired Behavior

For Cloudbeaver to start normally

### CloudBeaver Version

25.2.1

### Additional context

Red Hat Openshift 4.19.13

@EvgeniaBzzz Hi! I can. I'm stuck on `25.2.1` because of `pg_dump` absence.
When I upgrade cloud-dbeaver to any fresher than `25.2.1` version (in example below: `25.3.4`), I get:

```bash
❯ k logs cloud-dbeaver-855f45d6d5-clmzn -n apps
Starting Cloudbeaver Server
2026-02-05 11:17:33.482 - Workspace path initialized: /opt/cloudbeaver/workspace
2026-02-05 11:17:33.764 - Platform not initialized: metadata folder may be not set
2026-02-05 11:17:33.870 - Starting instance server at http://localhost:40035
SLF4J(I): Logback configuration is used: conf/logback.xml
05-02-2026 11:17:35.769 [main] DEBUG i.c.model.app.BaseServletApplication - Loading configuration from /opt/cloudbeaver/conf/cloudbeaver.conf
05-02-2026 11:17:35.771 [main] DEBUG i.c.s.CBServerConfigurationController - Using configuration [/opt/cloudbeaver/conf/cloudbeaver.conf]
05-02-2026 11:17:35.771 [main] DEBUG i.c.s.CBServerConfigurationController - Read configuration [/opt/cloudbeaver/conf/cloudbeaver.conf]
05-02-2026 11:17:36.171 [main] DEBUG i.c.s.CBServerConfigurationController - Runtime configuration [/opt/cloudbeaver/workspace/.data/.cloudbeaver.runtime.conf]
05-02-2026 11:17:36.172 [main] DEBUG i.c.s.CBServerConfigurationController - Read configuration [/opt/cloudbeaver/workspace/.data/.cloudbeaver.runtime.conf]
05-02-2026 11:17:36.270 [main] INFO  io.cloudbeaver.server.CBPlatform - Initialize web platform CBPlatform
05-02-2026 11:17:37.565 [main] DEBUG o.j.d.runtime.SecurityProviderUtils - BounceCastle bundle found. Use JCE provider BC
05-02-2026 11:17:37.768 [main] DEBUG o.j.d.registry.BasePlatformImpl - Initialize base platform...
05-02-2026 11:17:37.870 [main] DEBUG i.c.server.BaseServletPlatform - Temp folder: /tmp/dbeaver/.work-data
05-02-2026 11:17:40.163 [main] DEBUG o.j.d.r.DataSourceProviderRegistry - Total database drivers: 126 (126)
05-02-2026 11:17:40.164 [main] INFO  io.cloudbeaver.server.CBPlatform - Web platform initialized (3893ms)
05-02-2026 11:17:40.167 [main] DEBUG io.cloudbeaver.server.CBApplication - CloudBeaver CE Server 25.3.4.202602020958 is starting
05-02-2026 11:17:40.168 [main] DEBUG io.cloudbeaver.server.CBApplication - 	OS: Linux 5.15.0-164-generic (amd64)
05-02-2026 11:17:40.168 [main] DEBUG io.cloudbeaver.server.CBApplication - 	Java version: 21.0.9 by Eclipse Adoptium (64bit)
05-02-2026 11:17:40.168 [main] DEBUG io.cloudbeaver.server.CBApplication - 	Install path: '/opt/cloudbeaver/server'
05-02-2026 11:17:40.168 [main] DEBUG io.cloudbeaver.server.CBApplication - 	Global workspace: 'file:/opt/cloudbeaver/workspace/'
05-02-2026 11:17:40.168 [main] DEBUG io.cloudbeaver.server.CBApplication - 	Memory available 39Mb/123Mb
05-02-2026 11:17:40.168 [main] DEBUG io.cloudbeaver.server.CBApplication - 	Content root: /opt/cloudbeaver/web
05-02-2026 11:17:40.168 [main] DEBUG io.cloudbeaver.server.CBApplication - 	Drivers storage: /opt/cloudbeaver/drivers
05-02-2026 11:17:40.170 [main] DEBUG io.cloudbeaver.server.CBApplication - 	Listen port: 8978 on all interfaces
05-02-2026 11:17:40.171 [main] DEBUG io.cloudbeaver.server.CBApplication - 	Base URI: /api/
05-02-2026 11:17:40.171 [main] DEBUG io.cloudbeaver.server.CBApplication - 	Global access server URL: https://db.home
05-02-2026 11:17:40.171 [main] DEBUG io.cloudbeaver.server.CBApplication - 	Production mode
05-02-2026 11:17:40.361 [main] DEBUG io.cloudbeaver.server.CBApplication - 	Local host addresses:
05-02-2026 11:17:40.363 [main] DEBUG io.cloudbeaver.server.CBApplication - 		10.42.0.251 (cloud-dbeaver-855f45d6d5-clmzn)
05-02-2026 11:17:40.363 [main] DEBUG io.cloudbeaver.server.CBApplication - 		0:0:0:0:0:0:0:1%lo (localhost)
05-02-2026 11:17:40.866 [main] ERROR i.c.registry.WebDriverRegistry - 	Driver 'yandex_clickhouse' is missing library 'ru.yandex.clickhouse:clickhouse-jdbc:RELEASE'
05-02-2026 11:17:40.962 [main] INFO  i.c.registry.WebDriverRegistry - Available drivers: ClickHouse,Databend,Db2 for LUW,Db2 for IBM i,DuckDB,Firebird,Hadoop / Apache Kyuubi,Trino,H2 Embedded,H2 Embedded V.2,MS SQL Server / SQL Server,MySQL,MariaDB,Oracle,PostgreSQL,SQLite,LibSQL
05-02-2026 11:17:41.468 [main] DEBUG i.c.s.auth.ReverseProxyConfigurator - Reverse proxy provider disabled, migration not needed
05-02-2026 11:17:41.669 [main] DEBUG i.c.service.security.db.CBDatabase - Initiate management database
05-02-2026 11:17:41.964 [main] DEBUG i.c.service.security.db.CBDatabase - 	Initiate connection pool with management database (PostgreSQL; jdbc:postgresql://postgres.apps.svc.cluster.local:5432/cloud_dbeaver)
05-02-2026 11:17:45.078 [main] INFO  o.j.d.e.p.b.JDBCDatabasePostgresBackupHandler - Command started: pg_dump --host postgres.apps.svc.cluster.local --port 5432 --blobs --verbose --file /opt/cloudbeaver/workspace/backup/cloud_dbeaver_backup_version_cloud_dbeaver25.zip --schema cloud_dbeaver
05-02-2026 11:17:45.161 [main] ERROR o.j.d.e.p.b.JDBCDatabasePostgresBackupHandler - Create backup is failed: Cannot run program "pg_dump": error=2, No such file or directory
05-02-2026 11:17:45.162 [main] WARN  o.j.d.m.sql.schema.SQLSchemaManager - CB_CE migration has been rolled back
05-02-2026 11:17:45.164 [main] ERROR io.cloudbeaver.server.CBApplication - Error initializing database
org.jkiss.dbeaver.DBException: Error updating management database schema
	at io.cloudbeaver.service.security.db.CBDatabase.initialize(CBDatabase.java:164)
	at io.cloudbeaver.service.security.EmbeddedSecurityControllerFactory.createAndInitDatabaseInstance(EmbeddedSecurityControllerFactory.java:90)
	at io.cloudbeaver.service.security.EmbeddedSecurityControllerFactory.createSecurityService(EmbeddedSecurityControllerFactory.java:54)
	at io.cloudbeaver.server.CBApplicationCE.createGlobalSecurityController(CBApplicationCE.java:70)
	at io.cloudbeaver.server.CBApplication.initializeSecurityController(CBApplication.java:468)
	at io.cloudbeaver.server.CBApplication.startServer(CBApplication.java:303)
	at io.cloudbeaver.model.app.BaseServletApplication.start(BaseServletApplication.java:206)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:219)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:149)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:115)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:467)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:298)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.invokeFramework(DBeaverLauncher.java:968)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.basicRun(DBeaverLauncher.java:680)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.run(DBeaverLauncher.java:1776)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.main(DBeaverLauncher.java:1749)
Caused by: org.jkiss.dbeaver.DBException: Internal database backup has failed
	at org.jkiss.dbeaver.model.sql.schema.SQLSchemaManager.doBackupDatabase(SQLSchemaManager.java:154)
	at org.jkiss.dbeaver.model.sql.schema.SQLSchemaManager.updateSchema(SQLSchemaManager.java:129)
	at org.jkiss.dbeaver.model.sql.db.InternalDB.updateSchema(InternalDB.java:149)
	at io.cloudbeaver.service.security.db.CBDatabase.initializeSchema(CBDatabase.java:222)
	at org.jkiss.dbeaver.model.sql.db.InternalDB.initSchema(InternalDB.java:110)
	at io.cloudbeaver.service.security.db.CBDatabase.initialize(CBDatabase.java:162)
	... 17 common frames omitted
Caused by: org.jkiss.dbeaver.DBException: Create backup is failed: Cannot run program "pg_dump": error=2, No such file or directory
	at org.jkiss.dbeaver.ext.postgresql.backup.JDBCDatabasePostgresBackupHandler.doBackup(JDBCDatabasePostgresBackupHandler.java:89)
	at org.jkiss.dbeaver.model.sql.schema.SQLSchemaManager.doBackupDatabase(SQLSchemaManager.java:151)
	... 22 common frames omitted
05-02-2026 11:17:45.166 [main] DEBUG io.cloudbeaver.server.CBApplication - Cloudbeaver Server is stopping
05-02-2026 11:17:45.171 [Framework stop - Equinox Container: c016e729-cf43-4c89-a00d-1a339c46e055] DEBUG io.cloudbeaver.server.CBPlatform - Shutdown web platform
05-02-2026 11:17:45.171 [Framework stop - Equinox Container: c016e729-cf43-4c89-a00d-1a339c46e055] DEBUG o.j.d.registry.BasePlatformImpl - Dispose navigator model
05-02-2026 11:17:45.463 [Framework stop - Equinox Container: c016e729-cf43-4c89-a00d-1a339c46e055] DEBUG io.cloudbeaver.server.CBPlatform - Shutdown completed in 290ms
```

It's sad though that the pod manages to appear as healthy before crash loop so old one being removed.

--- a/plugins/org.jkiss.dbeaver.model.sql.jdbc/src/org/jkiss/dbeaver/model/sql/schema/SQLSchemaManager.java
+++ b/plugins/org.jkiss.dbeaver.model.sql.jdbc/src/org/jkiss/dbeaver/model/sql/schema/SQLSchemaManager.java
@@ -1,6 +1,6 @@
 /*
  * DBeaver - Universal Database Manager
- * Copyright (C) 2010-2025 DBeaver Corp and others
+ * Copyright (C) 2010-2026 DBeaver Corp and others
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -21,6 +21,7 @@ import org.jkiss.code.NotNull;
 import org.jkiss.code.Nullable;
 import org.jkiss.dbeaver.DBException;
 import org.jkiss.dbeaver.Log;
+import org.jkiss.dbeaver.model.app.DBPApplication;
 import org.jkiss.dbeaver.model.connection.InternalDatabaseConfig;
 import org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCTransaction;
 import org.jkiss.dbeaver.model.impl.sql.BasicSQLDialect;
@@ -30,6 +31,7 @@ import org.jkiss.dbeaver.model.sql.SQLDialect;
 import org.jkiss.dbeaver.model.sql.backup.JDBCDatabaseBackupDescriptor;
 import org.jkiss.dbeaver.model.sql.backup.JDBCDatabaseBackupRegistry;
 import org.jkiss.dbeaver.model.sql.translate.SQLQueryTranslator;
+import org.jkiss.dbeaver.runtime.DBWorkbench;
 import org.jkiss.dbeaver.utils.ContentUtils;
 import org.jkiss.utils.CommonUtils;
 import org.jkiss.utils.IOUtils;
@@ -151,6 +153,11 @@ public final class SQLSchemaManager {
                     descriptor.getInstance().doBackup(dbCon, currentSchemaVersion, databaseConfig);
                     log.info("Starting backup execution");
                 } catch (DBException e) {
+                    DBPApplication application = DBWorkbench.getPlatform().getApplication();
+                    if (application.isHeadlessMode() && application.isCommunity()) {
+                        log.warn("Database backup failed: " + e.getMessage());
+                        return;
+                    }
                     throw new DBException("Internal database backup has failed", e);
                 }
             }

